app/models/concerns/hostext/ownership.rb

```diff
@@ -0,0 +1,67 @@
+module Hostext
+  module Ownership
+    extend ActiveSupport::Concern
+
+    included do
+      OWNER_TYPES = %w(User Usergroup)
+      belongs_to :owner, :polymorphic => true
+
+      before_validation :set_default_user
+      validates :owner_type, :inclusion =>
+                               {:in => OWNER_TYPES,
+                               :allow_blank => true,
+                               :message     => (_("Owner type needs to be one of the following: %s") % OWNER_TYPES.join(', '))}
+      validate :validate_owner
+      validate :owner_taxonomies_match, :if => Proc.new { |host| host.owner.is_a?(User) }
+    end
+
+    # method to return the correct owner list for host edit owner select dropbox
+    def is_owned_by
+      owner.id_and_type if owner
+    end
+
+    # virtual attributes which sets the owner based on the user selection
+    # supports a simple user, or a usergroup
+    # selection parameter is expected to be an ActiveRecord id_and_type method (see Foreman's AR extentions).
+    def is_owned_by=(selection)
+      oid = User.find(selection.to_i) if selection =~ (/-Users\Z/)
+      oid = Usergroup.find(selection.to_i) if selection =~ (/-Usergroups\Z/)
+      self.owner = oid
+    end
+
+    private
+
+    def owner_taxonomies_match
+      return true if self.owner.admin?
+
+      if Organization.organizations_enabled && self.organization_id && !self.owner.my_organizations.where(:id => self.organization_id).exists?
+        errors.add :is_owned_by, _("does not belong into host's organization")
+      end
+      if Location.locations_enabled && self.location_id && !self.owner.my_locations.where(:id => self.location_id).exists?
+        errors.add :is_owned_by, _("does not belong into host's location")
+      end
+    end
+
+    def set_default_user
+      return if self.owner_type.present? && (!OWNER_TYPES.include?(self.owner_type) || self.owner.nil?)
+      owner_id = Setting[:host_owner]
+      owner, self.owner_type = User.find_by_id(owner_id.to_i), 'User' if self.owner_type.nil? && owner_id =~ (/-Users\Z/)
+      owner, self.owner_type = Usergroup.find_by_id(owner_id.to_i), 'Usergroup' if self.owner_type.nil? && owner_id =~ (/-Usergroups\Z/)
+      self.owner ||= owner || User.current
```

Comment:
You could extract this to a `HostOwnerSuggestion`-service. This would reduct complexity in the host's model and might be easier to test. Also this can be reused in the view. In addition it's more easily extendable.
I could imagine a use case where the default owner should be more dynamic, e.g. the Usergroup of the current user or similar. But that's something for a separate PR. :-)